person.rb

```diff
@@ -1,20 +1,23 @@
 class Person
-    attr_accessor :id, :name, :age
-    def initialize(age, name = 'unknown', parent_permission = true)
-        @age = age
-        @name = name
-        @parent_permission = parent_permission
-    end
-    def can_use_services?
-        is_of_age? && @parent_permission
-    end
-    def is_of_age?
-        @age >= 18
-    end
-    private :is_of_age?
+  attr_accessor :id, :name, :age
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
```

Comment:
- [ ] We want to set the `id` attribute to **read-only** since we don't want the `id` of each _Person_ to be **unique** and **constant**. 
- [ ] Again, we want to auto-generate a **unique** `id` for a _Person_ at the point at which the _Person_ is created.  Therefore we want to initialize the `id` during the point when a _Person_ class is created, like below:

```ruby
   attr_accessor :name, :age 
   attr_reader :id 

  def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
  ```